ClinGen/clincoded
# Connects to #868. Connects to #872. Adding evaluation forms

#### Connects to #868 
Added evaluation forms to rest of tabs based on wireframes and tab locations spreadsheet.

*Testing:*

1. Get to VCI
2. Confirm that correct evaluation forms for correct criteria exists under each tab
  * Confirm that the disease-dependent evaluations are disabled
  * Confirm that form groups w/ only disease-dependent criteria have their Save button disabled
  * Confirm that form groups w/ not only disease-dependent criteria have their Save button enabled, but only saves data for the disease-independent evaluations on Save
3. Add disease to interpretation
  * Confirm that disease-dependent evaluations are enabled

#### Connects to #872
Renamed dropdown items as specified in ticket. Added updated dropdown option populating logic.

*Testing:*

1. Get to VCI
2. For Benign criteria, dropdown options should have Met, Not Met, Not Evaluated, along with _P, _S, and _stand alone
3. For Pathogenic criteria, dropdown options should have Met, Not Met, Not Evaluated, along with _P, _M, _S, _VS, except for the one that matches the criteria (ie PM1 should not have _M, PS1 should not have _S, etc)
4. Attempt to save multiple criteria with non-Not Met/Not Evaluated should thrown an error.

#### Other changes
Tabs and subtabs now toggle the entire DOM rendering of their content, instead of using a CSS class to hide their visibility. Despite the worse performance this was done to reset the forms and form messages after tab switching (desired behavior).

*Testing:*

1. Switch tabs and subtabs, confirm that the tab switches work properly.
2. Change form values, but do not save. Switch tabs. Switch back. Values should have reset.
3. Change form values to bad values (multiple Mets, etc). Attempt Save. Switch tabs. Switch back. Values should have reset and error message should have disappeared.
4. Change form values and successfully save. Switch tabs. Switch back. Values should be the ones that were saved, with no more successful save message.

Forms now check for whether or not a previous interpretation exists. If it does, the Save button should be replaced with a light-blue Update button

![image](https://cloud.githubusercontent.com/assets/4326866/17578467/d0d8e064-5f3d-11e6-88b0-9218956991df.png)

*Testing:*
1. Get to VCI on new interpretation
2. Confirm that buttons are all blue 'Save'
3. Save evaluation. Confirm that the button changed to light-blue 'Update'
4. Switch tabs and switch back. Confirm that the button is still light-blue 'Update'
5. Re-save (Edit) evaluation. Confirm that button is still light-blue 'Update'
